fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/FluidStorage.java

```diff
@@ -47,11 +57,55 @@
 	public static final BlockApiLookup<Storage<FluidVariant>, Direction> SIDED =
 			BlockApiLookup.get(new Identifier("fabric:sided_fluid_storage"), Storage.asClass(), Direction.class);
 
+	/**
+	 * Item access to fluid variant storages.
+	 * Fluid amounts are always expressed in {@linkplain FluidConstants droplets}.
+	 *
+	 * <p>This may be queried both client-side and server-side.
+	 * Returned APIs should behave the same regardless of the logical side.
+	 */
+	// TODO: link the event and base implementations, and explain which wrappers Fabric API offers by default.
+	public static final ItemApiLookup<Storage<FluidVariant>, ContainerItemContext> ITEM =
+			ItemApiLookup.get(new Identifier("fabric:fluid_storage"), Storage.asClass(), ContainerItemContext.class);
+
+	/**
+	 * Allows multiple API providers to provide a {@code Storage<FluidVariant>} implementation for the same item.
+	 *
+	 * <p>When the item is queried for an API through {@link #ITEM}, all registered event handlers will be invoked.
+	 * All {@code Storage<FluidVariant>} passed to the {@code storageAdder} in the event handlers will be combined in a single storage,
+	 * that will be the final result of the query, or {@code null} if no storage is offered by the event handlers.
+	 *
+	 * <p>This is appropriate to use when multiple mods could wish to expose the Fluid API for some items,
+	 * for example when dealing with items added by the base Minecraft game such as buckets or empty bottles.
+	 * A typical usage example is a mod adding support for filling empty bottles with a honey fluid:
+	 * Fabric API already registers a storage for empty bottles to allow filling them with water through the event,
+	 * and a mod can register an event handler that will attach a second storage allowing empty bottles to be filled with its honey fluid.
+	 *
+	 * @throws IllegalStateException If an incompatible provider is already registered for the item.
+	 */
+	public static Event<FindForItem> findForItem(Item item) {
```

Comment:
With this there isn't really a good way to filter based on tags, or class or other. filters gooder and best in game

Reply:
Perhaps I can add a fallback event to have the same flexibility as API Lookup itself? Zero additional effort! 😆

Reply:
Done. Maybe this is worth adding to the api api.

Reply:
Update: not worth it cause the `ContainerItemContext` allows dropping the `ItemStack` parameter for this particular use case.